examples/chip-tool-darwin/commands/clusters/WriteAttributeCommandBridge.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <app/WriteClient.h>
+
+#include "ModelCommandBridge.h"
+
+class WriteAttribute : public ModelCommand {
+public:
+    WriteAttribute()
+        : ModelCommand("write-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("attribute-id", 0, UINT32_MAX, &mAttributeId);
+        AddArgument("attribute-value", &mAttributeValue);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        ModelCommand::AddArguments();
+    }
+
+    WriteAttribute(chip::ClusterId clusterId)
+        : ModelCommand("write-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("attribute-id", 0, UINT32_MAX, &mAttributeId);
+        AddArgument("attribute-value", &mAttributeValue);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        ModelCommand::AddArguments();
+    }
+
+    WriteAttribute(const char * _Nonnull attributeName)
+        : ModelCommand("write")
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+    }
+
+    ~WriteAttribute() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+        NSDictionary * valueDict = @{@"type" : @"Structure", @"value" : @[]};
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        ReturnErrorOnFailure(mAttributeValue.Encode(writer, chip::TLV::AnonymousTag()));
+        reader.Init(mData, writer.GetLengthWritten());
+        ReturnErrorOnFailure(reader.Next());
+
+        id value = NSObjectFromCHIPTLV(&reader);
+        if (value != nil) {
```

Comment:
Again, seems like if `value` is nil we should error out instead of doing a write with an empty value... (which might itself error out lower down, of course)

Reply:
Updated.